dftimewolf/lib/collectors/grr_base.py

```diff
@@ -61,13 +61,17 @@ def _check_approval_wrapper(self, grr_object, grr_function, *args, **kwargs):
       The return value of the execution of grr_function(*args, **kwargs).
     """
     approval_sent = False
-    object_id = getattr(grr_object, 'hunt_id') or getattr(grr_object, 'flow_id')
+    object_id = getattr(grr_object, 'hunt_id', None)
+    if not object_id:
+      object_id = getattr(grr_object, 'flow_id', None)
+    if not object_id:
+      object_id = getattr(grr_object, 'client_id', None)
 
     while True:
       try:
         return grr_function(*args, **kwargs)
-      except grr_errors.AccessForbiddenError:
-        print('{0:s}: no valid approval found'.format(object_id))
+      except grr_errors.AccessForbiddenError as err:
```

Comment:
Please use "as exception", per the style guide.

Reply:
Changed all other `except` clauses to match this